harvard-lts/jstor_aspace_oai_harvester
# Ss 129

**Harvester component in tempate form**
* * *

**JIRA Ticket**: [SS-129](https://jira.huit.harvard.edu/browse/SS-129)
* Parent jira ticket](https://jira.huit.harvard.edu/browse/SS-121)

# What does this Pull Request do?
Initial checkin including these features
- celery and rabbit mq connectivity
- - listener and do task model
- flask
- local docker
- test to submit message to queue

# How should this be tested?
Described in readme, copied below:
clone repository
- cp .env.example to .env
- cp celeryconfig.py.example to celeryconfig.py and put in credentials (from lastpass)
- make sure logs/jstor_harvester directory exists
- bring up docker
- - docker-compose -f docker-compose-local.yml up --build -d --force-recreate
- run test script to send hello world message to first queue
- - if needed install celery for the test script > pip install -U celery
- - > python3 ./tests/invoke-task.py
- - view rabbitmq ui, watch hello world message progress to second_queue (you can view the message)

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No unit tests for initial template
- integration tests?  In development in separate itests docker

# Interested parties
